server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/InMemoryTreeNode.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.node;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+import java.util.Map;
+
+import static org.opensearch.index.compositeindex.datacube.startree.utils.StarTreeUtils.ALL;
+
+/**
+ * /**
+ * Represents a node in a tree data structure, specifically designed for a star-tree implementation.
+ * A star-tree node will represent both star and non-star nodes.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class InMemoryTreeNode {
```

Comment:
Why have we not removed utils.TreeNode ?

Reply:
yeah it was removed in the file formats PR originally. Have removed it here as well!